modules/cockroachdb/options.go

```diff
@@ -1,69 +1,122 @@
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		for _, arg := range req.Cmd {
+			if arg == insecureFlag {
+				return errInsecureWithPassword
+			}
+		}
 
-// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
-func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
-	// NOOP to satisfy interface.
-	return nil
-}
+		req.Env[envPassword] = password
 
-// WithDatabase sets the name of the database to use.
-func WithDatabase(database string) Option {
-	return func(o *options) {
-		o.Database = database
+		return nil
 	}
 }
 
-// WithUser creates & sets the user to connect as.
-func WithUser(user string) Option {
-	return func(o *options) {
-		o.User = user
+// WithStoreSize sets the amount of available [in-memory storage].
+//
+// [in-memory storage]: https://www.cockroachlabs.com/docs/stable/cockroach-start#store
+func WithStoreSize(size string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		for i, cmd := range req.Cmd {
+			if strings.HasPrefix(cmd, memStorageFlag) {
+				req.Cmd[i] = memStorageFlag + size
+				return nil
+			}
+		}
+
+		// Wasn't found, add it.
+		req.Cmd = append(req.Cmd, memStorageFlag+size)
+
+		return nil
 	}
 }
 
-// WithPassword sets the password when using password authentication.
-func WithPassword(password string) Option {
-	return func(o *options) {
-		o.Password = password
+// WithNoClusterDefaults disables the default cluster settings script.
+//
+// Without this option Cockroach containers run `data/cluster-defaults.sql` on startup
+// which configures the settings recommended by Cockroach Labs for [local testing clusters]
+// unless data exists in the `/cockroach/cockroach-data` directory within the container.
+//
+// [local testing clusters]: https://www.cockroachlabs.com/docs/stable/local-testing
+func WithNoClusterDefaults() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		for i, file := range req.Files {
+			if _, ok := file.Reader.(*defaultsReader); ok && file.ContainerFilePath == clusterDefaultsContainerFile {
+				req.Files = append(req.Files[:i], req.Files[i+1:]...)
+				return nil
+			}
+		}
+
+		return nil
 	}
 }
 
-// WithStoreSize sets the amount of available in-memory storage.
-// See https://www.cockroachlabs.com/docs/stable/cockroach-start#store
-func WithStoreSize(size string) Option {
-	return func(o *options) {
-		o.StoreSize = size
+// WithInitScripts adds the given scripts to those automatically run when the container starts.
+// These will be ignored if data exists in the `/cockroach/cockroach-data` directory within the container.
+func WithInitScripts(scripts ...string) testcontainers.CustomizeRequestOption {
```

Comment:
Readers provides more flexibility so switching to that would be a good idea. Users that want to read from a file can easily use `embed` + `bytes.NewReader` to achieve that.

Reply:
Just came to implement this and we missed that you need a file name, which determines the order, so this would either require a custom type or fixed order and generate names, thoughts?

Reply:
We already have a custom type: ContainerFile, which holds a reader and a target file path:
```go
type ContainerFile struct {
	HostFilePath      string    // If Reader is present, HostFilePath is ignored
	Reader            io.Reader // If Reader is present, HostFilePath is ignored
	ContainerFilePath string
	FileMode          int64
}
```

Reply:
IMO that wouldn't add much value as you could just use that now. If we used `ContainerFile` for this the only benefit to having this method would be not having to set the destination directory.